### PR TITLE
Simplify API routes MessagingRP

### LINKED ISSUE #6269: Simplify API Registration - DatastoresRP, DaprRP, MessagingRP
## Overview of feature request

Ref: https://github.com/radius-project/radius/pull/5851

@youngbupark and @rynowak  helped update current service, httproutes, operations and api registration for corerp. 
The remaining RPs (DatastoresRP, DaprRP, MessagingRP) will need to have code updated to conform to updated api registration and merging of 'applications.datastores' service into now updated 'radius' service.

## Acceptance criteria
Code updated in DatastoresRP, DaprRP, MessagingRP for updated API registration using as reference updates done in pkg/corerp/setup and associated files in corerp. 
Single service 'radius' will serve all resource providers in radius.

## Additional context

[AB#9362](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/9362)

### PR DESCRIPTION
**Pull request by sk593**
# Description

Updating API routes for messagingrp

## Type of change

- This pull request adds or changes features of Radius and has an approved issue #6269 .

Fixes: partially fixes #6269 

### Summary
🔄🚀🗑️

This pull request refactors the codebase to move the messaging functionality from the `portableresources` resource provider to the `applications` resource provider. It adds new files for the `messagingrp` package, which defines the operations and setup for the messaging resource provider. It also updates the configuration files and the documentation to reflect the new port and resource provider for the messaging feature. It removes the obsolete code and tests related to messaging from the `portableresources` package.

> _`portableresources`_
> _No longer handles messaging_
> _Spring cleaning is done_

### Walkthrough
*  Add messaging resource provider to applications resource provider ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-667f05aa16f6dfd1d55d0ca1392411a09981560434524889a6dfe768679673fcR46), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-667f05aa16f6dfd1d55d0ca1392411a09981560434524889a6dfe768679673fcR161), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-b30407415dd2a84746f6d50dd44a5cbad71eb44630df88c735e715edfb852ac4R1-R84), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-d5a626f9259dd5b3c41f58ce71fdc82afd0c54be8a9835bf80e1a54fa2011f04R1-R92))
  * Import messagingrp/setup package in `cmd/applications-rp/main.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-667f05aa16f6dfd1d55d0ca1392411a09981560434524889a6dfe768679673fcR46))
  * Add messaging namespace builder to applications resource provider builder in `cmd/applications-rp/main.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-667f05aa16f6dfd1d55d0ca1392411a09981560434524889a6dfe768679673fcR161))
  * Define setup function for messaging namespace in `pkg/messagingrp/setup/setup.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-b30407415dd2a84746f6d50dd44a5cbad71eb44630df88c735e715edfb852ac4R1-R84))
  * Register messaging resource types and operations with builder and async job controller in `pkg/messagingrp/setup/setup.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-b30407415dd2a84746f6d50dd44a5cbad71eb44630df88c735e715edfb852ac4R1-R84))
  * Define operations for messaging resource provider in `pkg/messagingrp/setup/operations.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-d5a626f9259dd5b3c41f58ce71fdc82afd0c54be8a9835bf80e1a54fa2011f04R1-R92))
* Update configuration files and documentation for messaging resource provider port changes ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-ceba0600c7e49ffd65b0a2fd7bf1798d9f6f6f531db64e051bb00ff29c7dcd93L42-R42), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-be9c0b61a26165c87503e667cec005520aea94da66f0dd1eedfff863efd266acL40-R40), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-c5a6e900bac29ec26476b731e62862ee9afb8f9f67225da0aa8fd1d052f8183fL235-R235))
  * Use port 8080 for both Applications.Core and Applications.Messaging resource providers in ucp-self-hosted-dev environment in `cmd/ucpd/ucp-self-hosted-dev.yaml` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-ceba0600c7e49ffd65b0a2fd7bf1798d9f6f6f531db64e051bb00ff29c7dcd93L42-R42))
  * Use port 5443 for Applications.Messaging resource provider in ucp deployment in `deploy/Chart/templates/ucp/configmaps.yaml` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-be9c0b61a26165c87503e667cec005520aea94da66f0dd1eedfff863efd266acL40-R40))
  * Update documentation for ucp configuration settings to reflect new port for Applications.Messaging resource provider in `docs/contributing/contributing-code/contributing-code-control-plane/configSettings.md` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-c5a6e900bac29ec26476b731e62862ee9afb8f9f67225da0aa8fd1d052f8183fL235-R235))
* Remove messaging operations and routes from portableresources resource provider ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-72c62c901d96d8731589a498aa8646bb01aeefafd5dbcc0b56781120b151b14aL78-L87), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-72c62c901d96d8731589a498aa8646bb01aeefafd5dbcc0b56781120b151b14aL148-L157), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-72c62c901d96d8731589a498aa8646bb01aeefafd5dbcc0b56781120b151b14aL178-L187), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-72c62c901d96d8731589a498aa8646bb01aeefafd5dbcc0b56781120b151b14aL258-L297), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-56d7c0ab8720204bebac59bab3fcfadd0edfac6944a88c61d3188285a57d3c80L42-L45), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-56d7c0ab8720204bebac59bab3fcfadd0edfac6944a88c61d3188285a57d3c80L80-R80), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-56d7c0ab8720204bebac59bab3fcfadd0edfac6944a88c61d3188285a57d3c80L98-L235))
  * Remove operation for reading messaging operations from `pkg/portableresources/frontend/handler/getoperations.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-72c62c901d96d8731589a498aa8646bb01aeefafd5dbcc0b56781120b151b14aL78-L87))
  * Remove operation for registering messaging resource provider from `pkg/portableresources/frontend/handler/getoperations.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-72c62c901d96d8731589a498aa8646bb01aeefafd5dbcc0b56781120b151b14aL148-L157))
  * Remove operation for unregistering messaging resource provider from `pkg/portableresources/frontend/handler/getoperations.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-72c62c901d96d8731589a498aa8646bb01aeefafd5dbcc0b56781120b151b14aL178-L187))
  * Remove operations for managing rabbitMQQueues from `pkg/portableresources/frontend/handler/getoperations.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-72c62c901d96d8731589a498aa8646bb01aeefafd5dbcc0b56781120b151b14aL258-L297))
  * Remove imports for messagingrp packages from `pkg/portableresources/frontend/handler/routes.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-56d7c0ab8720204bebac59bab3fcfadd0edfac6944a88c61d3188285a57d3c80L42-L45))
  * Remove call to AddMessagingRoutes function from `pkg/portableresources/frontend/handler/routes.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-56d7c0ab8720204bebac59bab3fcfadd0edfac6944a88c61d3188285a57d3c80L80-R80))
  * Remove definition of AddMessagingRoutes function from `pkg/portableresources/frontend/handler/routes.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-56d7c0ab8720204bebac59bab3fcfadd0edfac6944a88c61d3188285a57d3c80L98-L235))
* Update and add unit tests for messaging resource provider and portableresources resource provider ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-d98a4356c6f2a78c5c9723a3306ff155039b5b2e69e156c91e738de0a7deb7aeR1-R88), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-935e98a3e7f642c3133e7514a00b6b326bc1a321b9a266fac3912ca49fbd4ca5L50-R50), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-2792dae138ce1d90b14d02d5f207e29a4a144886ca9ece6faab8e8effc47e851L31), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-2792dae138ce1d90b14d02d5f207e29a4a144886ca9ece6faab8e8effc47e851L38-L65), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-2792dae138ce1d90b14d02d5f207e29a4a144886ca9ece6faab8e8effc47e851L221-L228), [link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-2792dae138ce1d90b14d02d5f207e29a4a144886ca9ece6faab8e8effc47e851L264-L270))
  * Add unit tests for setup function for messaging namespace in `pkg/messagingrp/setup/setup_test.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-d98a4356c6f2a78c5c9723a3306ff155039b5b2e69e156c91e738de0a7deb7aeR1-R88))
  * Update expected number of operations in test for getoperations.go in `pkg/portableresources/frontend/handler/getoperations_test.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-935e98a3e7f642c3133e7514a00b6b326bc1a321b9a266fac3912ca49fbd4ca5L50-R50))
  * Remove import for messagingrp controller from `pkg/portableresources/frontend/handler/routes_test.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-2792dae138ce1d90b14d02d5f207e29a4a144886ca9ece6faab8e8effc47e851L31))
  * Remove test cases for messaging routes from `pkg/portableresources/frontend/handler/routes_test.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-2792dae138ce1d90b14d02d5f207e29a4a144886ca9ece6faab8e8effc47e851L38-L65))
  * Remove test cases for messaging operation statuses and results from `pkg/portableresources/frontend/handler/routes_test.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-2792dae138ce1d90b14d02d5f207e29a4a144886ca9ece6faab8e8effc47e851L221-L228))
  * Remove test case for messaging operations from `pkg/portableresources/frontend/handler/routes_test.go` ([link](https://github.com/radius-project/radius/pull/6586/files?diff=unified&w=0#diff-2792dae138ce1d90b14d02d5f207e29a4a144886ca9ece6faab8e8effc47e851L264-L270))

### CODE REVIEW

File: pkg/messagingrp/setup/setup.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	pr_ctrl "github.com/radius-project/radius/pkg/portableresources/backend/controller"
+	rp_frontend "github.com/radius-project/radius/pkg/rp/frontend"
+)
+
+const (
+	// AsyncOperationRetryAfter is polling interval for async create/update or delete resource operations.
+	AsyncOperationRetryAfter = time.Duration(5) * time.Second
+)
+
+// SetupNamespace builds the namespace for datastores resource provider.
```
Comment (ytimocin) [inline pkg/messagingrp/setup/setup.go:38]:
```suggestion
// SetupNamespace builds the namespace for messaging resource provider.
```

**Review by ytimocin — DISMISSED** [review]

**Review by vishwahiremat — DISMISSED** [review]

**Review by lakshmimsft — DISMISSED** [review]

File: pkg/messagingrp/setup/setup.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+// SetupNamespace builds the namespace for messaging resource provider.
+func SetupNamespace(recipeControllerConfig *controllerconfig.RecipeControllerConfig) *builder.Namespace {
+	ns := builder.NewNamespace("Applications.Messaging")
+
+	_ = ns.AddResource("rabbitMQQueues", &builder.ResourceOption[*datamodel.RabbitMQQueue, datamodel.RabbitMQQueue]{
+		RequestConverter:  converter.RabbitMQQueueDataModelFromVersioned,
+		ResponseConverter: converter.RabbitMQQueueDataModelToVersioned,
+
+		Put: builder.Operation[datamodel.RabbitMQQueue]{
```
Comment (ytimocin) [inline pkg/messagingrp/setup/setup.go:46]: I see that Put and Patch are identical most of the time and we can just have one definition (like var) for this and pass it into both Put and Patch.

Comment (ytimocin) [reply pkg/messagingrp/setup/setup.go:46]: We can file an issue for this.

**Review by ytimocin — APPROVED** [review]

**Pull request merged**